catpol/spiders/cdep_career_history.py

```diff
@@ -0,0 +1,44 @@
+import scrapy
+import logging
+
+import sys
+from scrapy.http import Response
+
+import catpol.loaders as loaders
+import catpol.items as items
+import catpol.http as http
+import string
+import catpol.cmdinput as cmdinput
+from typing import Sequence, TypeVar, Coroutine
+
+
+class CdepCareerHistory(scrapy.Spider):
+    name = 'cdep_career_history'
+
+    def __init__(self, after=None, years=None):
+        super().__init__(after=None, years=None)
+        self.years = cmdinput.expand_years(after, years)
+
+    def start_requests(self) -> Coroutine:
+        for page in list(self.revert_pagination(self.years)):
+            yield http.Reqo(url=page, callback=self.get_politician_urls)
+
+    def revert_pagination(self, years: list) -> list:
+        return ['http://www.cdep.ro/pls/parlam/structura2015.de?leg={}&par=A{}'.format(year, char)
+                for year in years
+                for char in string.ascii_uppercase]
+
+    def get_politician_urls(self, response: Response) -> Coroutine:
```

Comment:
I do not understand what you want to say with
> Therefore, not being implemented generator as a data type, I used the wider coroutine (codomain of my function).

I feel like we're arguing noodles here.
```
>>> def f():
...     for i in range(10):
...         yield i
... 
>>> type(f())
<class 'generator'>
```

It's true—
```
>>> asyncio.coroutines.iscoroutine(f())
True
```
but it is beyond the point. PEP 484 (Type Hints) [says that](https://www.python.org/dev/peps/pep-0484/#annotating-generator-functions-and-coroutines) we should annotate these type of functions as such—`Generator[yield_type, send_type, return_type]`. Since Python does not do any kind of type checking, I think we should stick to the PEP recommendation, rather than bringing our own opinions to the table.